### PR TITLE
[Meta]: Add sonar cloud scanner

### PR DESCRIPTION
**Pull request by ad3154**
Added github action for scanning the repo with sonar cloud

### CODE REVIEW

**Comment by ad3154** [comment]
Hmmm... either the code is really good, or it doesn't produce much until it runs on the default branch...

File: sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1,14 @@
+sonar.projectKey=ad3154_ISO11783-CAN-Stack
+sonar.organization=isobus-plusplus
+
+# This is the name and version displayed in the SonarCloud UI.
+#sonar.projectName=ISO11783-CAN-Stack
+#sonar.projectVersion=1.0
+
+# Path is relative to the sonar-project.properties file. Replace "\" by "/" on Windows.
+sonar.sources=isobus/src,examples,hardware_integration,utility
```
Comment (GwnDaan) [inline sonar-project.properties:9]:
```suggestion
sonar.sources=isobus,examples,hardware_integration,utility

# Exclude some files from the analysis.
sonar.exclusions=**/PCANBasic.h
```

**Review by GwnDaan — CHANGES_REQUESTED** [review]
Looks good for now I think. I believe the current CLI won't produce 'coverage' results as described here [linux-cmake-gcovr-gh-actions-sc](https://github.com/sonarsource-cfamily-examples/linux-cmake-gcovr-gh-actions-sc), but we can add that later, the percentage of code coverage won't be high anyway I'm afraid 😅

**Comment by GwnDaan** [comment]
> Hmmm... either the code is really good, or it doesn't produce much until it runs on the default branch...

Yea.. I think the code analysis on a PR only takes into account the code that has changed, which for this PR is none. When we merge it, it will tell us a lot more, at least I hope so.

**Review by GwnDaan — APPROVED** [review]

**Pull request merged**